### PR TITLE
Real usage

### PR DESCRIPTION
**Pull request by pinin4fjords**
This PR improves the operation of schema generation to get us closer to information currently in the controlled vocabulary table: https://docs.google.com/spreadsheets/d/1avk8-0NJ1Qt-iE4Fu7VOpH_XiPadSQ919VB04aoJ__E/edit#gid=0. 

Changes are:

- Create a dictionary of ontology URIs to reference. The names can then be used in the schema generation file.
- Use OWL files rather than obo (they seem to be more easily available).
- Make error reporting useful and actionable

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,117 +17,158 @@ pip install -e .
...
+Here is an example:
 
 ```
-column1	obo:https://www.ebi.ac.uk/efo/efo.obo
-column2	[human, mouse, rat, non-human primate, other]
-column3	obo:https://raw.githubusercontent.com/pato-ontology/pato/master/pato.obo	PATO:0001894
+field    type    controlled_vocab        use_na    parent_id    use_ids
+column1    string    None        False    NA    NA
+column2    integer    None        False    NA    NA
+column3    string    ['value1', 'value2', 'value3']    False    NA    NA
```
Comment (luxeredias) [inline README.md:52]:
```suggestion
level    field    type    example_value    controlled_vocab        use_na    parent_id    use_ids
level1    column1    string    string1    None        False    NA    NA
level1    column2    integer    100    None        False    NA    NA
level2    column3    string    value1    ['value1', 'value2', 'value3']    False    NA    NA
```

Comment (luxeredias) [reply README.md:52]: I know this is just an example, so if you feel like it's ok to keep it simpler, no need to accept the suggestion (including the one before for the fields required in the TSV input file for the schema generation.

**Review by luxeredias — APPROVED** [review]
Everything is working well - both generate-schema and validate-csv.

I was able to create the schema you propose from example_schema_generation_input.tsv and use validate-csv on a mock csv input I created from the fields in the schema. 

Should we add this mock example? If so, I have to create a dedicated PR do to that, right?

File: README.md [hunk 2]
```diff
@@ -17,117 +17,158 @@ pip install -e .
...
-The following is an example of an input file (`example_schema_generation_input.tsv`) that can be used to generate a schema:
+1. `field`: The name of the field.
+2. `type`: The data type of the field. Valid values are "string", "number", "integer", and "boolean".
+3. `controlled_vocab`: The controlled vocabulary for the field. This can be one of the following:
+   - An ontology name from the [ONTOLOGY_LIST](csv_validator/ontology_list.py).
+   - A comma-separated list of allowed values enclosed in square brackets (e.g., "[value1, value2, value3]").
+   - "None" or an empty string if there is no controlled vocabulary for the field.
+4. `use_na` (optional): Whether the field allows the value "NA". If set to "true", the field will be marked as nullable.
+5. `parent_id` (optional): The ID of the parent term in the ontology to filter terms. Use "None" if not applicable.
+6. `use_ids` (optional): Whether to use ontology term IDs instead of names. If set to "true", the term IDs will be used.
```
[the remark this answers is not in the document]
Comment (pinin4fjords) [reply README.md:44]: No probs- I'll add a workaround to convert ':' to '_'

File: README.md (hunk 1 above)
Comment (pinin4fjords) [reply README.md:52]: The example is intentionally minimal. There's nothing to stop you adding other fields in 'real life' (they will simply be ignored), but they don't need to be here.

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (pinin4fjords) [reply README.md:44]: Again, the additional columns do not contribute to validation, so should not be added here and muddy the message.

However the new commit https://github.com/deeplifeai/csv_validator/pull/3/commits/e80f0691fb1548858b82055eb948e58167c8b617 will ensure that any additional columns will be added into the schema to inform any annotators that end up reading it.

**Comment by pinin4fjords** [comment]
Thanks for the review @luxeredias I've added commits that address your points so merging now. Happy to refine from here of course.

**Pull request merged**